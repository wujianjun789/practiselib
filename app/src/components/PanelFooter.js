import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
/**
 * Panel footer with button
 * @prop funcNames {array}  btn's function
 * @prop btnTitles {array}  btn's title
 */
export default class PanelFooter extends PureComponent {
    render() {
        const {funcNames, btnTitles, btnDisabled=[false, false], btnClassName=['btn-default','btn-primary']} = this.props;
        return (
            <div className="row">
                <div className="btn-toolbar pull-right padding-right" role="toolbar" aria-label="toolbar">
                    <div className="btn-group" role='group' aria-label="group">
                        <button type="button" className={`btn ${btnClassName[0]}`} disabled={btnDisabled[0]} onClick={this.props[funcNames[0]]}>{btnTitles[0]}</button>
                    </div>
                    <div className="btn-group" role="group" aria-label="group">
                        <button type="button" className={`btn ${btnClassName[1]}`} disabled={btnDisabled[1]} onClick={this.props[funcNames[1]]}>{btnTitles[1]}</button>
                    </div>
                </div>
            </div>
        )
    }
}

PanelFooter.PropTypes = {
    funcNames: PropTypes.arrayOf(PropTypes.string).isRequired,
    btnTitles: PropTypes.arrayOf(PropTypes.string).isRequired,
    btnDisabled: PropTypes.arrayOf(PropTypes.bool)
}