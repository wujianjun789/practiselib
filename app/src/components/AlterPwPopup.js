import React, {PureComponent} from 'react';
import Panel from './Panel';
import PanelFooter from './PanelFooter';

export default class AlterPwPopup extends PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            oldPw: '',
            newPw: '',
            repPw: ''
        }
        this.onCancel = this.onCancel.bind(this);
        this.onConfirm = this.onConfirm.bind(this);
        this.onChange = this.onChange.bind(this);
    }

    onCancel() {
        this.props.overlayerHide();
    }

    onConfirm() {
        this.props.overlayerHide();
        console.log(this.state);
    }

    onChange(e) {
        let id = e.target.id;
        this.setState({[id]: e.target.value});
    }

    render() {
        let {className=''} = this.props;
        let footer = <PanelFooter funcNames={['onCancel','onConfirm']} btnTitles={['取消','确认']} btnClassName={['btn-default', 'btn-primary']} btnDisabled={[false, false]} onCancel={this.onCancel} onConfirm={this.onConfirm}/>;
        return <div className={className}>
            <Panel className='panel-custom'  title='修改密码'  footer={footer} closeBtn={true} closeClick={this.onCancel}>
                <div className="form-group clearfix">
                    <label htmlFor="oldPw" className="col-sm-3 control-label">旧密码：</label>
                    <div className="col-sm-9">
                        <input type="password" className="form-control" id="oldPw" placeholder="输入旧密码" value={this.state.oldPw} onChange={this.onChange}/>
                    </div>
                </div>
                <div className="form-group clearfix">
                    <label htmlFor="newPw" className="col-sm-3 control-label">新密码：</label>
                    <div className="col-sm-9">
                        <input type="password" className="form-control" id="newPw" placeholder="输入新密码" value={this.state.newPw} onChange={this.onChange}/>
                    </div>
                </div>
                <div className="form-group clearfix">
                    <label htmlFor="repPw" className="col-sm-3 control-label">重复密码：</label>
                    <div className="col-sm-9">
                        <input type="password" className="form-control" id="repPw" placeholder="再次输入新密码" value={this.state.repPw} onChange={this.onChange}/>
                    </div>
                </div>
            </Panel>
        </div>
    }
}