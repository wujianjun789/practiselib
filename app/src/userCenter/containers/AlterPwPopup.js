import React, {PureComponent} from 'react';
import Panel from '../../components/Panel';
import PanelFooter from '../../components/PanelFooter';

export default class AlterPwPopup extends PureComponent {
    constructor(props) {
        super(props);
    }

    render() {
        let {className=''} = this.props;
        let footer = <PanelFooter funcNames={['onCancel','onConfirm']} btnTitles={['取消','确认']} btnClassName={['btn-default', 'btn-primary']} btnDisabled={[false, true]} />;
        return <div className={className}>
            <Panel className='panel-custom' title='修改密码' footer={footer} closeBtn={true} closeClick={()=>this.closeClick()}>
                <div className="form-group clearfix">
                    <label htmlFor="oldPw" className="col-sm-3 control-label">旧密码：</label>
                    <div className="col-sm-9">
                        <input type="text" className="form-control" id="oldPw" placeholder="输入旧密码" />
                    </div>
                </div>
                <div className="form-group clearfix">
                    <label htmlFor="newPw" className="col-sm-3 control-label">新密码：</label>
                    <div className="col-sm-9">
                        <input type="text" className="form-control" id="newPw" placeholder="输入新密码" />
                    </div>
                </div>
                <div className="form-group clearfix">
                    <label htmlFor="repPw" className="col-sm-3 control-label">重复密码：</label>
                    <div className="col-sm-9">
                        <input type="text" className="form-control" id="repPw" placeholder="再次输入新密码" />
                    </div>
                </div>
            </Panel>
        </div>
    }
}