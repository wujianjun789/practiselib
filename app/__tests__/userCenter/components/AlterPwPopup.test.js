import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import AlterPwPopup from '../../../src/components/AlterPwPopup';
import PanelFooter from '../../../src/components/PanelFooter';

describe('<AlterPwPopup />', () => {
    it('default render', () => {
        const cmp = shallow(<AlterPwPopup />);

        let panel = cmp.find('Panel');
        expect(panel.length).toBe(1);
        expect(panel.hasClass('panel-custom')).toBeTruthy();
        expect(panel.props().title).toBe('修改密码');
        expect(panel.props().closeBtn).toBeTruthy();

        let oldInput = cmp.find('#oldPw');
        expect(oldInput.prop('placeholder')).toBe('输入旧密码');
        expect(oldInput.props().value).toBe('');
        oldInput.simulate('change', {target: {value: 'oldPw', id: 'oldPw'}});
        oldInput = cmp.find('#oldPw');
        expect(oldInput.props().value).toBe('oldPw');

        let newInput = cmp.find('#newPw');
        expect(newInput.prop('placeholder')).toBe('输入新密码');
        expect(newInput.props().value).toBe('');
        newInput.simulate('change', {target: {value: 'newPw', id: 'newPw'}});
        newInput = cmp.find('#newPw');
        expect(newInput.props().value).toBe('newPw');

        let repInput = cmp.find('#repPw');
        expect(repInput.prop('placeholder')).toBe('再次输入新密码');
        expect(repInput.props().value).toBe('');
        repInput.simulate('change', {target: {value: 'repPw', id: 'repPw'}});
        repInput = cmp.find('#repPw');
        expect(repInput.props().value).toBe('repPw');
    });

    it('renderer with className="alter-pw"', () => {
        const cmp = shallow(<AlterPwPopup className="alter-pw"/>);
        expect(cmp.find('.alter-pw').length).toBe(1);
    });

    it('snapshot', () => {
        const cmp = renderer.create(<AlterPwPopup />);
        const tree = cmp.toJSON();
        expect(tree).toMatchSnapshot();
    });
})