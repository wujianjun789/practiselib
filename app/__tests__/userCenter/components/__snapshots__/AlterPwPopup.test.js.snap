exports[`<AlterPwPopup /> snapshot 1`] = `
<div
  className="">
  <div
    className="modal "
    role="dialog">
    <div
      className="modal-dialog"
      role="document">
      <div
        className="modal-content">
        <div
          className="modal-header">
          <button
            aria-label="Close"
            className="close"
            data-dismiss="modal"
            onClick={[Function]}
            type="button">
            <span
              aria-hidden="true">
              ×
            </span>
          </button>
          <h4
            className="modal-title">
            修改密码
          </h4>
        </div>
        <div
          className="modal-body">
          <div
            className="form-group row">
            <label
              className="col-sm-3 control-label"
              htmlFor="oldPw">
              旧密码：
            </label>
            <div
              className="col-sm-9">
              <input
                className="form-control"
                id="oldPw"
                onChange={[Function]}
                placeholder="输入旧密码"
                type="password"
                value="" />
            </div>
          </div>
          <div
            className="form-group row">
            <label
              className="col-sm-3 control-label"
              htmlFor="newPw">
              新密码：
            </label>
            <div
              className="col-sm-9">
              <input
                className="form-control"
                id="newPw"
                onChange={[Function]}
                placeholder="输入新密码"
                type="password"
                value="" />
            </div>
          </div>
          <div
            className="form-group row">
            <label
              className="col-sm-3 control-label"
              htmlFor="repPw">
              重复密码：
            </label>
            <div
              className="col-sm-9">
              <input
                className="form-control"
                id="repPw"
                onChange={[Function]}
                placeholder="再次输入新密码"
                type="password"
                value="" />
            </div>
          </div>
        </div>
        <div
          className="modal-footer">
          <button
            className="btn btn-default"
            disabled={false}
            onClick={[Function]}
            type="button">
            取消
          </button>
          <button
            className="btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button">
            确认
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
